entities/markets/marketCondition.ts

```diff
@@ -0,0 +1,30 @@
+import { Market } from '@/entities/markets/market';
+import { Condition as ConditionalTokenCondition } from '@/queries/conditional-tokens/types';
+
+export class MarketCondition {
```

Comment:
Not sure about it, there are some potential issues:
- it would create a tight coupling between Market and MarketCondition.
- MarketCondition would inherit all properties and methods from Market, which is not very relevant. (only isWinner)
- It could lead to confusion about whether a particular instance represents a market or a market condition.

the good part
- it removes the need to receive a market model and can just receive the `fpmm` as the MarketModel
- calling isWinner would be more simple, `this.isWinner`

Reply:
After talking with Leo, I converted MarketCondition to extend Market, as we already pass a market there there's already a relationship so we can leverage the extend and make it more powerful and simple.